### PR TITLE
CVSB-3946 - throw error on missing TIR cert number

### PR DESCRIPTION
**Pull request by KHodderDVSA**

### CODE REVIEW

File: src/services/TestResultsService.ts [hunk 1]
```diff
@@ -429,18 +431,26 @@ export class TestResultsService {
       });
   }
 
-  public isMissingRequiredCertificateNumberOnAdr(payload: ITestResultPayload): boolean {
+  private isMissingRequiredCertificateNumber(typeFunc: (arg1: any) => boolean, payload: ITestResultPayload) {
```
Comment (agoncearenco) [inline src/services/TestResultsService.ts]: I like the solution! Can you just rename `arg1` to `testType` ?
Also, don't we have an interface for the testType?

File: tests/unit/insertTestResult.unitTest.ts [hunk 2]
```diff
@@ -83,6 +83,47 @@ describe("insertTestResult", () => {
...
+
+            testResultsService = new TestResultsService(new MockTestResultsDAO());
+            for (const testType of mockData.testTypes) {
+                delete testType.testCode;
+                delete testType.testNumber;
+                delete testType.lastUpdatedAt;
+                delete testType.testAnniversaryDate;
+                delete testType.createdAt;
+                delete testType.testExpiryDate;
+                delete testType.certificateLink;
```
Comment (agoncearenco) [inline tests/unit/insertTestResult.unitTest.ts]: Is it relevant to this test to remove all these fields?

File: src/services/TestResultsService.ts (hunk 1 above)
Comment (agoncearenco) [reply src/services/TestResultsService.ts]: Thank you!

Comment (KHodderDVSA) [reply src/services/TestResultsService.ts]: revised

File: tests/unit/insertTestResult.unitTest.ts (hunk 2 above)
Comment (KHodderDVSA) [reply tests/unit/insertTestResult.unitTest.ts]: yeah, annoyingly.

**Pull request merged**